earth_enterprise/src/fusion/dbmanifest/dbmanifest_unittest.cpp

```diff
@@ -145,21 +250,310 @@ TEST_F(DbManifestTest, EnforcesAbsolutePath) {
+  mock_get_index_manifest->manifest_[1].data_size = 1024;
+  mock_get_index_manifest->manifest_[2].orig_path = khComposePath(disconnect_path, assets_root_, "SomeDataDir/somedatafile3");
+  mock_get_index_manifest->manifest_[2].current_path = mock_get_index_manifest->manifest_[0].orig_path;
+  mock_get_index_manifest->manifest_[2].data_size = 1024;
+
+  expected_stream_manifest = mock_get_index_manifest->manifest_;
+  expected_stream_manifest[0].orig_path = expected_stream_manifest[0].orig_path.substr(disconnect_path.length());
+  expected_stream_manifest[1].orig_path = expected_stream_manifest[1].orig_path.substr(disconnect_path.length());
+  expected_stream_manifest[2].orig_path = expected_stream_manifest[2].orig_path.substr(disconnect_path.length());
+  expected_stream_manifest.
+    push_back(ManifestEntry(db_create_params.poi_entries[0].poi_file,
+    khComposePath(disconnect_path, db_create_params.poi_entries[0].poi_file)));
+  expected_search_manifest.
+    push_back(ManifestEntry(db_create_params.poi_entries[0].poi_file,
+    khComposePath(disconnect_path, db_create_params.poi_entries[0].poi_file)));
+  expected_stream_manifest.back().dependents.
+    push_back(ManifestEntry(db_create_params.poi_entries[0].poi_data_files[0],
+    khComposePath(disconnect_path, db_create_params.poi_entries[0].poi_data_files[0])));
+  expected_search_manifest.back().dependents.
+    push_back(ManifestEntry(db_create_params.poi_entries[0].poi_data_files[0],
+    khComposePath(disconnect_path, db_create_params.poi_entries[0].poi_data_files[0])));
+  expected_stream_manifest.push_back(ManifestEntry(toc_path_, khComposePath(disconnect_path, toc_path_)));
+  expected_stream_manifest.push_back(ManifestEntry(db_header_path_, khComposePath(disconnect_path, db_header_path_)));
+  expected_stream_manifest.
+    push_back(ManifestEntry(db_create_params.icon_files[0],
+    khComposePath(disconnect_path, db_create_params.icon_files[0])));
+  expected_stream_manifest.
+    push_back(ManifestEntry(db_create_params.icon_files[1],
+    khComposePath(disconnect_path, db_create_params.icon_files[1])));
+  expected_stream_manifest.
+    push_back(ManifestEntry(db_create_params.icon_files[2],
+    khComposePath(disconnect_path, db_create_params.icon_files[2])));
+  expected_stream_manifest.
+    push_back(ManifestEntry(db_create_params.icon_files[3],
+    khComposePath(disconnect_path, db_create_params.icon_files[3])));
+
+  // change pointer to base type
+  get_index_manifest = TransferOwnership(mock_get_index_manifest);
+
+  EXPECT_NO_THROW(db_manifest = TransferOwnership(new DbManifest(&db_path, get_index_manifest)));
+  EXPECT_EQ(db_path_v1_, db_path); // should be changed now
+  EXPECT_EQ(db_manifest->Prefix(), disconnect_path);
+
+  db_manifest->GetPushManifest(file_pool_, &stream_manifest, &search_manifest, "", "");
+
+  bool is_equivalent = IsEquivalent(&expected_stream_manifest, &stream_manifest);
+  if (!is_equivalent) {
```

Comment:
should this fail the test case ?

Reply:
Same as above.  Line 538 in this case will do the failure.

Reply:
ok